index.js

```diff
@@ -0,0 +1,82 @@
+const express = require('express');
+const db = require('./data/dbConfig');
+const mw = require('./middleware/middleware');
+const bcrypt = require('bcryptjs');
+const session = require('express-session');
+const KnexSessionStore = require('connect-session-knex')(session);
+
+const server = express();
+server.use(express.json());
+
+const PORT = 4070;
+const sessionConfig = {
+  name: 'mySession',
+  secret: 'This is not a secret!',
+  cookie: {
+    secure: false,
+    maxAge: 1000 * 60 * 10
+  },
+  store: new KnexSessionStore({
+    tablename: 'sessions',
+    sidfieldname: 'sid',
+    knex: db,
+    createtable: true,
+    clearInterval: 1000 * 60 * 60
+  }),
+  httpOnly: true,
+  resave: false,
+  saveUninitialized: false
+};
+
+server.use(session(sessionConfig));
+
+server.post('/api/register', (req, res) => {
+  const user = req.body;
+  if (!user.password || !user.username) {
+    res.status(400).json({ error: 'Username and Password are required!' });
+  }
+  user.password = bcrypt.hashSync(user.password, 12);
+  db('users')
+    .insert(user)
+    .then(ids => res.status(201).json(ids[0]))
```

Comment:
You could [destructure](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Destructuring_assignment#Array_destructuring) the first element in your `then` callback:
```javascript
.then(([ id ]) => res.status(201).json(id))
```